edx_solutions_api_integration/users/views.py

```diff
@@ -761,14 +761,17 @@ def post(self, request, user_id):  # pylint: disable=R0915
                 for old_image_size, old_image_name in profile_image_names.items():
                     if storage.exists(old_image_name):
                         try:
-                            file = storage.open(old_image_name)
-                            storage.save(new_profile_image_names[old_image_size], ContentFile(file.read()))
-                        except OSError:
+                            image_file = storage.open(old_image_name)
+                        except IOError:
```

Comment:
the `except` clauses are unnecessary if you are just raising the exceptions. I think you can just enclose the save storage code in try-catch and return an error message

```python
try:
    storage.save(new_profile_image_names[old_image_size], ContentFile(image_file.read()))
except IOError:
    response_data['message'] = _('Could not update profile image')
    return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
else:
   storage.delete(old_image_name)
```

Reply:
Done